php-checks/src/main/java/org/sonar/php/checks/phpunit/AssertTrueInsteadOfDedicatedAssertCheck.java

```diff
@@ -0,0 +1,166 @@
+    NOT_SAME("NotSame", "An object reference comparison"),
+    EQUALS("Equals", "An equals check"),
+    NOT_EQUALS("NotEquals", "An equals check");
+
+    public final String methodName;
+    public final String actionDescription;
+    public final String useInsteadMessage;
+    public final String secondaryExplanationMessage;
+
+    ReplacementAssertion(String namePostfix, String actionDescription) {
+      this.methodName = "assert" + namePostfix;
+      this.actionDescription = actionDescription;
+      this.useInsteadMessage = String.format(MESSAGE, methodName);
+      this.secondaryExplanationMessage = String.format(SECONDARY_MESSAGE, actionDescription, methodName);
+    }
+  }
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree fct) {
+    if (isPhpUnitTestMethod()) {
+      Optional<Assertion> assertion = getAssertion(fct);
+      assertion.ifPresent(a -> { if (ASSERT_METHOD_NAMES.contains(a.name())) { checkBooleanExpressionInAssertMethod(fct, a.name()); }});
+    }
+    super.visitFunctionCall(fct);
+  }
+
+  private void checkBooleanExpressionInAssertMethod(FunctionCallTree problematicAssertionCall, String assertionName) {
+    ExpressionTree argumentExpression = problematicAssertionCall.arguments().get(0);
+    Optional<ReplacementAssertion> replacementAssertionOpt = getReplacementAssertion(argumentExpression);
+    if (assertionName.equals("assertFalse")) {
+      replacementAssertionOpt = replacementAssertionOpt.map(COMPLEMENTS::get);
+    }
+
+    replacementAssertionOpt.ifPresent(replacementAssertion -> reportIssue(problematicAssertionCall, replacementAssertion, argumentExpression));
+  }
+
+  private void reportIssue(FunctionCallTree problematicAssertionCall, ReplacementAssertion replacementAssertion, ExpressionTree argumentExpression) {
+    context().newIssue(this, problematicAssertionCall, replacementAssertion.useInsteadMessage)
+      .secondary(argumentExpression, replacementAssertion.secondaryExplanationMessage);
+  }
+
+  /**
+   * Returns the assertX method that should be used instead of assertTrue, if applicable.
+   *
+   * @param argumentExpression the boolean expression passed to assertTrue
+   * @return the assertion method to be used instead of assertTrue, or {@code null} if no better assertion method was determined
+   */
+  private static Optional<ReplacementAssertion> getReplacementAssertion(ExpressionTree argumentExpression) {
+    ReplacementAssertion replacementAssertion = null;
+    argumentExpression = skipParenthesis(argumentExpression);
+
+    switch (argumentExpression.getKind()) {
+      case EQUAL_TO:
+        if (isCheckForNull((BinaryExpressionTree) argumentExpression)) {
+          replacementAssertion = ReplacementAssertion.NULL;
+        } else {
+          replacementAssertion = ReplacementAssertion.EQUALS;
+        }
+        break;
+      case STRICT_EQUAL_TO:
+        if (isPositionVerification((BinaryExpressionTree) argumentExpression)) {
```

Comment:
I'm not sure I get the reason for this (I've read the comment below). What is the problem in suggesting `assertSame(false, strpos(....))` instead of `assertTrue(strpos(....) === false)` ?